Get rid of required fields for npm task

damccorm opened the pull request:
From #198 none of these should actually be required.

Fixes #198

vtbassmatt commented:
You have to update the [schema generator](https://github.com/Microsoft/azure-pipelines-vscode/blob/master/src/schemata.ts) and run it, checking in the results, in order to not fail the build. (Yes, this is lame and is the reason we're replacing with your endpoint :grin:)

damccorm commented:
Oh, good, that's definitely a better way to do this. I updated accordingly and not surprisingly caught many more similar issues.

This is also good because it means it has parity with what I've introduced on the server side.
